tripal/includes/tripal.registration.inc

```diff
@@ -0,0 +1,370 @@
+      '#date_year_range' => '-20:+20',
+      '#date_format' => 'Y',
+    );
+  }
+
+  $form['details']['funding']['add_funding'] = array(
+    '#type' => 'button',
+    '#value' => t('Add additional funding sources'),
+    '#href' => '',
+    '#ajax' => array(
+      'callback' => 'custom_registration_ajax_add_funding',
+      'wrapper' => 'funding',
+    ),
+  );
+
+  // Provide a submit button.
+  if (!empty($form_data)) {
+    $form['submit'] = array(
+      '#type' => 'submit',
+      '#value' => 'Update registration information',
+    );
+  }
+  else {
+    $form['submit'] = array(
+      '#type' => 'submit',
+      '#value' => 'Register Site',
+    );
+  }
+
+
+  return $form;
+
+}
+
+function custom_registration_ajax_add_funding($form, $form_state) {
+  return $form['details']['funding'];
+}
+
+function _tripal_form_disable_reg_callback($form, &$form_state) {
+  variable_set('disable_tripal_reporting', TRUE);
+}
+
+/**
+ * Implements validation from the Form API.
+ *
+ * @param $form
+ *   A structured array containing the elements and properties of the form.
+ * @param $form_state
+ *   An array that stores information about the form's current state
+ *   during processing.
+ */
+function tripal_registration_form_validate($form, &$form_state){
+  $mail_pi = $form_state['values']['principal_investigator_email'];
+  $mail_sa = $form_state['values']['tripal_reg_site_admin_email'];
+  if ($form_state['values']['disable_tripal_reporting'] != TRUE) {
+    if (!empty($mail_pi) && !valid_email_address($mail_pi)) {
+      form_set_error('[details][principal_investigator][principal_investigator_email]', t('The email address for the principal investigator appears to be invalid.'));
+    }
+    if(!empty($mail_sa) && !valid_email_address($mail_sa)) {
+      form_set_error('[details][site_admin][tripal_reg_site_admin_email]', t("The email address for the site administrator appears to be invalid."));
+    }
+  }
+}
+
+/**
+ * Implements submit from the Form API.
+ *
+ * @param $form
+ *   A structured array containing the elements and properties of the form.
+ * @param $form_state
+ *   An array that stores information about the form's current state
+ *   during processing.
+ */
+function tripal_registration_form_submit($form, &$form_state) {
+  variable_set('disable_tripal_reporting', TRUE);
+
+  //Check for empty funding periods and remove them.
+  $j = 1;
+  foreach ($form_state['values']['funding'] as $funding_source) {
+    if (!empty($funding_source['tripal_reg_site_agency']) && !empty($funding_source['tripal_reg_site_grant'])) {
+      $form_state['values']['fundings'][$j] = $funding_source;
+      $j++;
+    }
+  }
+  $form_state['values']['funding'] = $form_state['values']['fundings'] ?? [];
+  $registration = serialize($form_state);
+  variable_set('tripal_site_registration', $registration);
+
+  //Now send the updated info to the Tripal Site.
+  tripal_registration_remote_submit($registration);
```

Comment:
This following should fix the problem I mentioned, ensuring that the site is only registered if they did not opt. out.

```suggestion
  // Only register with tripal.info if the user has not opt'd out.
  if ($form_state['values']['disable_tripal_reporting'] == FALSE) {
    tripal_registration_remote_submit($registration);
    drupal_set_message(t('Registration sent to tripal.info'), 'status');
  }
  else {
    drupal_set_message(t('You are not registered with tripal.info'), 'status');
  }
```